lib/gateway.js

```diff
@@ -0,0 +1,476 @@
+      operation: 'query',
+      name: {
+        kind: Kind.NAME,
+        value: 'EntitiesQuery'
+      },
+      variableDefinitions: [{
+        kind: Kind.VARIABLE_DEFINITION,
+        variable: {
+          kind: Kind.VARIABLE,
+          name: {
+            kind: Kind.NAME,
+            value: 'representations'
+          }
+        },
+        type: {
+          kind: Kind.NON_NULL_TYPE,
+          type: {
+            kind: Kind.LIST_TYPE,
+            type: {
+              kind: Kind.NON_NULL_TYPE,
+              type: {
+                kind: Kind.NAMED_TYPE,
+                name: {
+                  kind: Kind.NAME,
+                  value: '_Any'
+                }
+              }
+            }
+          }
+        },
+        directives: []
+      }],
+      directives: [],
+      selectionSet: {
+        kind: Kind.SELECTION_SET,
+        selections: [{
+          kind: Kind.FIELD,
+          name: {
+            kind: Kind.NAME,
+            value: '_entities'
+          },
+          arguments: [
+            {
+              kind: Kind.ARGUMENT,
+              name: {
+                kind: Kind.NAME,
+                value: 'representations'
+              },
+              value: {
+                kind: Kind.VARIABLE,
+                name: {
+                  kind: Kind.NAME,
+                  value: 'representations'
+                }
+              }
+            }
+          ],
+          directives: [],
+          selectionSet: {
+            kind: Kind.SELECTION_SET,
+            selections: [
+              {
+                kind: Kind.FIELD,
+                name: {
+                  kind: Kind.NAME,
+                  value: '__typename'
+                },
+                arguments: [],
+                directives: []
+              },
+              {
+                kind: Kind.INLINE_FRAGMENT,
+                typeCondition: {
+                  kind: Kind.NAMED_TYPE,
+                  name: {
+                    kind: Kind.NAME,
+                    value: inlineFragmentOnType
+                  }
+                },
+                directives: [],
+                selectionSet: {
+                  kind: Kind.SELECTION_SET,
+                  selections
+                }
+              }
+            ]
+          }
+        }]
+      }
+    }]
+  }
+}
+
+function createFieldResolverOperation (fragmentType, fieldName, selections) {
+  return createEntityReferenceResolverOperation(fragmentType, [{
+    kind: Kind.FIELD,
+    name: {
+      kind: Kind.NAME,
+      value: fieldName
+    },
+    directives: [],
+    selectionSet: {
+      kind: Kind.SELECTION_SET,
+      selections
+    }
+  }])
+}
+
+function makeResolveQuery (service) {
+  return function (root, args, context, info) {
+    const { fieldName, returnType, fieldNodes } = info
+    const type = getNamedType(returnType)
+
+    let selections = JSON.parse(JSON.stringify(fieldNodes[0].selectionSet.selections))
```

Comment:
please use http://npm.im/rdfc if possible (if not put a comment that it is not really possible)

Reply:
I tried and it is not possible. The fields array is not a simple object and it throws a max call stack size exceeded error.

Reply:
If it's not a simple object this should not really work.

Reply:
It was possible to remove this in the lastest change